xdsl/dialects/llvm.py

```diff
@@ -105,6 +105,43 @@ def is_typed(self):
         return not isinstance(self.type, NoneAttr)
 
 
+@irdl_attr_definition
+class LLVMArrayType(ParametrizedAttribute, MLIRType):
+    name = "llvm.array"
+
+    size: ParameterDef[IntAttr]
+    type: ParameterDef[Attribute]
+
+    def print_parameters(self, printer: Printer) -> None:
+        printer.print_string("<")
+        printer.print_string(str(self.size.data))
+        printer.print_string(" x ")
+        printer.print_attribute(self.type)
+        printer.print_string(">")
+
+    @staticmethod
+    def parse_parameters(parser: BaseParser) -> list[Attribute]:
+        if not parser.tokenizer.starts_with('<'):
+            return [NoneAttr(), NoneAttr()]
+        parser.parse_characters('<', "llvm.array parameters expected")
+        size = IntAttr(parser.parse_int_literal())
+        if not parser.tokenizer.starts_with('x'):
+            parser.parse_characters('>', "End of llvm.array type expected!")
+            return [size, NoneAttr()]
+        parser.parse_characters(
+            'x', "llvm.array size and type must be separated by `x`")
+        type = parser.try_parse_type()
+        if type is None:
+            parser.raise_error(
+                "Expected second parameter of llvm.array to be a type!")
+        parser.parse_characters('>', "End of llvm.array parameters expected!")
+        return [size, type]
+
+    @staticmethod
+    def from_size_and_type(size: IntAttr, type: Attribute):
```

Comment:
If you want, you can make `size` here accept `int | IntAttr` to make using it easier (and in-line with the other `.from...` methods)

Reply:
Done :)